test/EFCore.PG.Tests/NpgsqlValueGeneratorSelectorTest.cs

```diff
@@ -19,128 +20,157 @@ public class NpgsqlValueGeneratorSelectorTest
+            var builder = NpgsqlTestHelpers.Instance.CreateConventionBuilder();
+            builder.Entity<AnEntity>(
+                b =>
+                {
+                    b.HasKey(e => e.Binary);
+                    b.Property(e => e.Binary).HasDefaultValueSql("Foo").ValueGeneratedOnAdd();
+                });
+            var model = builder.FinalizeModel();
             var entityType = model.FindEntityType(typeof(AnEntity));
 
             var selector = NpgsqlTestHelpers.Instance.CreateContextServices(model).GetRequiredService<IValueGeneratorSelector>();
 
-            Assert.Equal(
-                CoreStrings.NoValueGenerator("Random", "AnEntity", typeof(Random).Name),
-                Assert.Throws<NotSupportedException>(() => selector.Select(entityType.FindProperty("Random"), entityType)).Message);
+            var generator = selector.Select(entityType.FindProperty("Binary"), entityType);
+            Assert.IsType<BinaryValueGenerator>(generator);
+            Assert.True(generator.GeneratesTemporaryValues);
         }
 
         [Fact]
-        public void Returns_generator_configured_on_model_when_property_is_Identity()
+        public void Returns_sequence_value_generators_when_configured_for_model()
         {
-            var model = NpgsqlTestHelpers.Instance.BuildModelFor<AnEntity>();
-            model.Npgsql().ValueGenerationStrategy = NpgsqlValueGenerationStrategy.SequenceHiLo;
-            model.Npgsql().GetOrAddSequence(NpgsqlModelAnnotations.DefaultHiLoSequenceName);
-            var entityType = model.FindEntityType(typeof(AnEntity));
-
-            var selector = NpgsqlTestHelpers.Instance.CreateContextServices(model).GetRequiredService<IValueGeneratorSelector>();
-
-            Assert.IsType<NpgsqlSequenceHiLoValueGenerator<int>>(selector.Select(entityType.FindProperty("Id"), entityType));
+            AssertGenerator<NpgsqlSequenceHiLoValueGenerator<int>>("Id", setSequences: true);
+            AssertGenerator<CustomValueGenerator>("Custom", setSequences: true);
+            AssertGenerator<NpgsqlSequenceHiLoValueGenerator<long>>("Long", setSequences: true);
+            AssertGenerator<NpgsqlSequenceHiLoValueGenerator<short>>("Short", setSequences: true);
+            AssertGenerator<NpgsqlSequenceHiLoValueGenerator<int>>("NullableInt", setSequences: true);
+            AssertGenerator<NpgsqlSequenceHiLoValueGenerator<long>>("NullableLong", setSequences: true);
+            AssertGenerator<NpgsqlSequenceHiLoValueGenerator<short>>("NullableShort", setSequences: true);
+            AssertGenerator<StringValueGenerator>("String", setSequences: true);
+            AssertGenerator<GuidValueGenerator>("Guid", setSequences: true);
+            AssertGenerator<BinaryValueGenerator>("Binary", setSequences: true);
         }
 
-        static IMutableModel BuildModel(bool generateValues = true)
+
+//        [ConditionalFact]
+//        public void Throws_for_unsupported_combinations()
```

Comment:
In theory yes, but porting this test is a bit more difficult since the upstream version depends on some stuff from the InMemory test project, which we don't have. Ideally we'd do this at some point, but I don't want to block on it right now.